tests/test_parser_steps.py

```diff
@@ -0,0 +1,600 @@
+            'No more diagonal lines',
+        ]
+        parser.step.get_steps(input_var)
+
+        # Then my input variable is not modified
+        self.assertEqual(
+            input_var,
+            [
+                'When I call get_step with a string like this',
+                '"""',
+                '    multiline string',
+                '"""',
+                '',
+                '# Test second step',
+                '',
+                'Then I am happy to see this step',
+                'No more diagonal lines',
+            ],
+        )
+
+    def test_get_steps_single_step(self):
+        """
+            Confirm we can get a single step when getting steps.
+        """
+        # Given I have Romaine core's parser
+        parser = common.get_romaine_parser()
+
+        # When I call get_steps on a list containing:
+        # """[
+        #     'When I call get_step with a table like this',
+        #     '|this table|'
+        # ] """
+        input_data = [
+            'When I call get_step with a table like this',
+            '|this table|',
+        ]
+        result = parser.step.get_steps(input_data)
+
+        # Then the result is a single step of type=When with
+        # text='I call get_step with a table like this', and a multiline_arg
+        # of type=table with [['this table']], and nothing remaining
+        self.assertEqual(
+            result,
+            {
+                'steps': [{
+                    'leading_comments_and_space': [],
+                    'type': 'When',
+                    'text': 'I call get_step with a table like this',
+                    'multiline_arg': {
+                        'type': 'table',
+                        'data': [['this table']],
+                    },
+                    'trailing_whitespace': [],
+                }],
+                'remaining': [],
+                'raw_input': input_data,
+            },
+        )
+
+    def test_get_two_steps(self):
+        """
+            Confirm we can get two steps.
+        """
+        # Given I have Romaine core's parser
+        parser = common.get_romaine_parser()
+
+        # When I call get_steps on a list containing:
+        # """[
+        #     '# Test first step
+        #     'When I call get_step with a string like this',
+        #     '"""'
+        #     '    multiline string'
+        #     '"""'
+        #     '',
+        #     '# Test second step',
+        #     '',
+        #     'Then I am happy not to see this as a step yet',
+        # ] """
+        input_data = [
+            'When I call get_step with a string like this',
+            '"""',
+            '    multiline string',
+            '"""',
+            '',
+            '# Test second step',
+            '',
+            'Then I am happy to see this step',
+            'No more diagonal lines',
+        ]
+        result = parser.step.get_steps(input_data)
+
+        # Then the result is two steps: type=When with
```

Comment:
See response regarding the other BDD style test comments.

I think these all need reworking, and wrapping around a bunch of functions which we could then use in both the 'traditional' tests and as the backend for the BDD ones.